kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/service/HostPort.java

```diff
@@ -86,7 +86,7 @@ public int port() {
      * @param address stringified form of the host port, separated by colon (:).
      * @return a {@link HostPort}
      */
-    @JsonCreator
+    @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
```

Comment:
so, is the change of behaviour that somehow older versions of Jackson were erroneously using delegating style to pass through the value to the constructor?

Where now it's inheriting the `.setConstructorDetector(ConstructorDetector.USE_PROPERTIES_BASED)` properties based default, so we need to override it.

Am I right that if this were properties based it would require input like `{ "address" : "host.name:1010"}`, just making sure I have the right idea of what PROPERTIES_BASED is vs DELEGATING.

Reply:
> so, is the change of behaviour that somehow older versions of Jackson were erroneously using delegating style to pass through the value to the constructor?

Yeah thats how I understand the discussion on https://github.com/FasterXML/jackson-databind/issues/4785 

> Where now it's inheriting the .setConstructorDetector(ConstructorDetector.USE_PROPERTIES_BASED) properties based default, so we need to override it.

Yes

> Am I right that if this were properties based it would require input like { "address" : "host.name:1010"}, just making sure I have the right idea of what PROPERTIES_BASED is vs DELEGATING.

I'm not clear on the difference but your understanding is the same as mine.